packages/protobuf/src/private/binary-format.ts

```diff
@@ -105,12 +105,15 @@ export function makeBinaryFormat(): BinaryFormat {
       let fieldNo: number | undefined, wireType: WireType | undefined;
       while (reader.pos < end) {
         [fieldNo, wireType] = reader.tag();
-        if (wireType == WireType.EndGroup) {
+        if (
+          (delimitedMessageEncoding ?? false) &&
```

Comment:
This will transpile to a relatively complex expression.

```suggestion
          delimitedMessageEncoding === true &&
```

It would also be fine to treat it as bool with a linter ignore like on line 101.